cos-gpu-installer-docker/entrypoint.sh

```diff
@@ -214,15 +225,46 @@ install_cross_toolchain_pkg() {
   info "Configuring environment variables for cross-compilation"
   export PATH="/build/bin:${PATH}"
   export SYSROOT="/build/usr/x86_64-cros-linux-gnu"
-  export CC="x86_64-cros-linux-gnu-gcc"
+}
+
+set_compilation_env() {
+  # Get toolchain_info path from COS GCS bucket
+  local -r tc_info_file_path="${COS_DOWNLOAD_GCS}/${BUILD_ID}/${TOOLCHAIN_INFO_FILENAME}"
+  info "Obtaining toolchain_info file from ${tc_info_file_path}"
+  local -r download_tc_info_url="$(curl -sfS "${tc_info_file_path}")"
+
+  # try to download toolchain_info if present
+  local attempts=0
+  until time curl -sfS "${download_tc_info_url}" -o "${TOOLCHAIN_INFO_FILENAME}"; do
+    attempts=$(( ${attempts} + 1 ))
+    if (( "${attempts}" >= "${RETRY_COUNT}" )); then
+      # Not able to download toolchain_info file from COS GCS bucket
+      break
+    fi
+    warn "Error fetching toolchain_info file from ${download_tc_info_url}, retrying"
+    sleep 1
+  done
+
+  if [[ -s ${TOOLCHAIN_INFO_FILENAME} ]]; then
+    # toolchain_info file will set 'toolchain_path', 'CC' and 'CXX'
+    # environment variable based on the toolchain used for kernel compilation
+    source "${TOOLCHAIN_INFO_FILENAME}"
+    local -r download_tc_url="${CHROMIUMOS_SDK_GCS}/${toolchain_path}"
+  else
+    # Required to support COS builds not having toolchain_info file
```

Comment:
Can we populate toolchain_info file for all COS images ever released so that we can get rid of the "else"?